src/Persistence/MassTransit.DapperIntegration/Configuration/Configuration_JobConsumers.cs

```diff
@@ -0,0 +1,288 @@
+
+            configurator.UseRepositoryRegistrationProvider(registrationProvider);
+
+            return configurator;
+        }
+    }
+
+    public interface IDapperJobSagaRepositoryConfigurator
+    {
+        /// <summary>
+        /// Override the context factory for the <see cref="JobSaga"/> repository.
+        /// </summary>
+        void UseJobContextFactory(Func<IServiceProvider, DatabaseContextFactory<JobSaga>> factoryFunc);
+
+        /// <summary>
+        /// Override the context factory for the <see cref="JobTypeSaga"/> repository.
+        /// </summary>
+        void UseJobTypeContextFactory(Func<IServiceProvider, DatabaseContextFactory<JobTypeSaga>> factoryFunc);
+
+        /// <summary>
+        /// Override the context factory for the <see cref="JobAttemptSaga"/> repository.
+        /// </summary>
+        void UseJobAttemptContextFactory(Func<IServiceProvider, DatabaseContextFactory<JobAttemptSaga>> factoryFunc);
+
+        /// <summary>
+        /// Configures the saga to use Microsoft SQL Server with this connection string.
+        /// Can also be set with <seealso cref="DapperOptions{TSaga}.ConnectionString"/> and <seealso cref="DapperOptions{TSaga}.Provider"/>.
+        /// </summary>
+        /// <param name="connectionString">The connection string to use</param>
+        void UseSqlServer(string connectionString);
+
+        /// <summary>
+        /// Configures the saga to use PostgreSQL with this connection string.
+        /// Can also be set with <seealso cref="DapperOptions{TSaga}.ConnectionString"/> and <seealso cref="DapperOptions{TSaga}.Provider"/>.
+        /// </summary>
+        /// <param name="connectionString">The connection string to use</param>
+        void UsePostgres(string connectionString);
+
+        /// <summary>
+        /// Configures the saga to use this transaction level.  Defaults to <see cref="System.Data.IsolationLevel.Serializable"/>.
+        /// Can also be set with <seealso cref="DapperOptions{TSaga}.IsolationLevel"/>.
+        /// </summary>
+        /// <param name="isolationLevel">The isolation level to use for all operations for this saga</param>
+        void UseIsolationLevel(IsolationLevel isolationLevel);
+    }
+
+    public class DapperJobSagaRepositoryConfigurator : IDapperJobSagaRepositoryConfigurator
+    {
+        Func<IServiceProvider, DatabaseContextFactory<JobSaga>>? _jobContextFactory;
+        Func<IServiceProvider, DatabaseContextFactory<JobTypeSaga>>? _jobTypeContextFactory;
+        Func<IServiceProvider, DatabaseContextFactory<JobAttemptSaga>>? _jobAttemptContextFactory;
+
+        protected DatabaseProviders Provider;
+        protected string? ConnectionString { get; set; }
+        protected string? TableName { get; set; }
+        protected IsolationLevel? IsolationLevel { get; set; }
+
+        public void UseJobContextFactory(Func<IServiceProvider, DatabaseContextFactory<JobSaga>> factoryFunc)
```

Comment:
MT does not use lambda method bodies in source. Follow the guidelines and use Rider's reformat and match syntax style to ensure you're code matches. Otherwise, I have to do it before merging.